wolfcrypt/src/port/nrf51.c

```diff
@@ -74,18 +74,12 @@ int nrf51_random_generate(byte* output, word32 size)
     }
 
     while (remaining > 0) {
-        err_code = nrf_drv_rng_bytes_available(&available);
-        if (err_code == NRF_SUCCESS) {
-            length = (remaining < available) ? remaining : available;
-            if (length > 0) {
-                err_code = nrf_drv_rng_rand(&output[pos], length);
-                remaining -= length;
-                pos += length;
-            }
-        }
-
-        if (err_code != NRF_SUCCESS) {
-            break;
+        nrf_drv_rng_bytes_available(&available);
```

Comment:
What is the reason for removing the return code checking here?

Reply:
The old API was:
`ret_code_t nrf_drv_rng_bytes_available(uint8_t * p_bytes_available);`
it is currently:
`void nrf_drv_rng_bytes_available(uint8_t * p_bytes_available);`
This API is present in both SDK v12.3 (newest SDK supporting the NRF51 series) and SDK v15.2 (newest SDK). The old API is still present in SDK v10. I should probably use a macro to choose which API to use but I'm not sure which one to use.